common/src/gas_provider/internal.rs

```diff
@@ -592,17 +526,19 @@ where
     /// message went to wait list, so wasn't consumed but the one generated
     /// during the execution of the original message went to message queue
     /// and was successfully executed.
-    fn consume(key: impl Into<GasNodeIdOf<Self>>) -> ConsumeResultOf<Self> {
+    fn consume(key: impl Into<Self::NodeId>) -> ConsumeResultOf<Self> {
         let key = key.into();
         let mut node = Self::get_node(key).ok_or_else(InternalError::node_not_found)?;
 
         if node.is_consumed() {
             return Err(InternalError::node_was_consumed().into());
         }
 
-        if !node.lock().is_zero() {
-            return Err(InternalError::consumed_with_lock().into());
-        }
+        // Check if at least one lock has not been released
+        ensure!(
```

Comment:
I'd remove foreign macros here to support the idea one day remove substrate deps here